### PR TITLE
Get unit tests building again.

### PR DESCRIPTION
**Pull request by kevingranade**
Reverse some of the bitrot of the minimal unit tests.
I'm looking into some header-only unit test frameworks to include so that people don't have to grab a dependency to build/run the unit tests.  Recommendations are welcome.
The front runner at the moment is https://github.com/philsquared/Catch

### CODE REVIEW

**Comment by kevingranade** [comment]
Refreshed again, this should build on any platform dda can build on in the first place since Catch is pulled into the repository, and it has some reasonable tests now.

File: tests/Makefile [hunk 1]
```diff
@@ -14,9 +14,10 @@ SOURCE_OBJS = $(patsubst %,../$(ODIR)/%,$(filter-out main.o,$(_OBJS)))
 
 ODIR := obj
 
-LDFLAGS += -L. -ltap++ -lboost_regex -lpthread
+LDFLAGS += -L.
```
Comment (BevapDin) [inline tests/Makefile]: On my system, I need to add `-lrt` to get `clock_gettime`.

**Comment by kevingranade** [comment]
Good to go I think, the player temperature test still doesn't work, but the right fix for that is breaking the player/game/map dependency thicket, and that's going to take some time.

**Comment by kevingranade** [comment]
Anybody?  I can guarantee it's not going to break anything >_<

**Comment by Rivet-the-Zombie** [comment]
I've held off on this because I'm not familiar enough with unit tests to properly mergetest it. If nobody else does so before then, I'll attempt to mergetest it in a few days.

**Comment by kevingranade** [comment]
If you use make, it's just:
make tests
tests/test_name1
Tests/test_name2
This reminds me, I should add a test build target to the C::B project file.

Slightly more depth:
The unit tests are micro-programs that run pieces of game code and check
that the results are what we expect, I pulled in a test framework to make
them a little easier to write.  You can write your own test harness and
test runner, but it's a pain.

Future plan
Ideally we start having decent coverage of parts of the code that change a
lot and/or are prone to breakage, and we can run them as a first step in
checking out a new PR and when we make our own changes.

**Pull request merged**